src/i965_gpe_utils.c

```diff
@@ -1406,6 +1406,27 @@ i965_object_surface_to_2d_gpe_resource(struct i965_gpe_resource *res,
 }
 
 void
+i965_object_surface_to_2d_gpe_resource_with_align(struct i965_gpe_resource *res,
+                                                  struct object_surface *obj_surface)
+{
+    unsigned int swizzle;
```

Comment:
For those special surface, it will be better directly change its orig_width and orig_height to 16 aligned value. So this aligned function is not required and  i965_object_surface_to_2d_gpe_resource() is enough for your requirement